api/v1alpha1/healthcheck_types.go

```diff
@@ -132,6 +132,16 @@ type ActiveHealthCheck struct {
 	// It's optional, and can only be used if the specified type is GRPC.
 	// +optional
 	GRPC *GRPCActiveHealthChecker `json:"grpc,omitempty" yaml:"grpc,omitempty"`
+
+	// RemoveAbsentHealthyEndpoints determines if endpoints that were previously discovered and are now absent from
```

Comment:
can we do without an API field ?
are there use cases where the API has removed the endpoint, but we want the data plane to keep using it ?

Reply:
The envoy rationale for this (default) behavior is outlined [here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery#on-eventually-consistent-service-discovery): 

>**Host absent / health check OK:**
Envoy will route to the target host. This is very important since the design assumes that the discovery service can fail at any time. If a host continues to pass health check even after becoming absent from the discovery data, Envoy will still route. Although it would be impossible to add new hosts in this scenario, existing hosts will continue to operate normally. When the discovery service is operating normally again the data will eventually re-converge.

Reply:
if the discovery service is not providing endpoints but DNS is, how will this setting help ?

Reply:
I tested out envoy's behavior when DNS resolution fails due to network connectivity issues. When IgnoreHealthOnHostRemoval is true, endpoints are not being dropped. I'm not sure if this covers all possible edge-cases for DNS resolution (e.g. NXDOMAIN, NODATA results), but I feel fine with only changing the default for the time being.